markets/perps-market/contracts/storage/PerpsCollateralConfiguration.sol

```diff
@@ -0,0 +1,158 @@
+     */
+    function loadValid(uint128 collateralId) internal view returns (Data storage collateralConfig) {
+        collateralConfig = load(collateralId);
+        if (collateralConfig.id == 0) {
+            revert InvalidId(collateralId);
+        }
+    }
+
+    /**
+     * @dev Load a valid  collateral LiquidationAssetManager configuration data using collateral/synth id
+     */
+    function loadValidLam(
+        uint128 collateralId
+    ) internal view returns (LiquidationAssetManager.Data storage collateralLAMConfig) {
+        collateralLAMConfig = load(collateralId).lam;
+        if (collateralLAMConfig.id == 0) {
+            revert InvalidId(collateralId);
+        }
+    }
+
+    function validDistributorExists(uint128 collateralId) internal view returns (bool) {
+        return
+            (collateralId == SNX_USD_MARKET_ID) ||
+            (loadValidLam(collateralId).distributor != address(0));
+    }
+
+    function setMax(Data storage self, uint128 collateralId, uint256 maxAmount) internal {
+        if (self.id == 0) self.id = collateralId;
+        self.maxAmount = maxAmount;
+    }
+
+    function setDiscounts(
+        Data storage self,
+        uint256 upperLimitDiscount,
+        uint256 lowerLimitDiscount,
+        uint256 discountScalar
+    ) internal {
+        self.upperLimitDiscount = upperLimitDiscount;
+        self.lowerLimitDiscount = lowerLimitDiscount;
+        self.discountScalar = discountScalar;
+    }
+
+    function getConfig(
+        Data storage self
+    )
+        internal
+        view
+        returns (
+            uint256 maxAmount,
+            uint256 upperLimitDiscount,
+            uint256 lowerLimitDiscount,
+            uint256 discountScalar
+        )
+    {
+        maxAmount = self.maxAmount;
+        upperLimitDiscount = self.upperLimitDiscount;
+        lowerLimitDiscount = self.lowerLimitDiscount;
+        discountScalar = self.discountScalar;
+    }
+
+    function isSupported(Data storage self) internal view returns (bool) {
+        return self.maxAmount != 0;
+    }
+
+    function valueInUsd(
+        Data storage self,
+        uint256 amount,
+        ISpotMarketSystem spotMarket,
+        PerpsPrice.Tolerance stalenessTolerance,
+        bool useDiscount
+    ) internal view returns (uint256 collateralValueInUsd, uint256 discount) {
+        uint256 skewScale = spotMarket.getMarketSkewScale(self.id);
+        // only discount collateral if skew scale is set on spot market and useDiscount is set to true
+        if (useDiscount && skewScale != 0) {
+            uint256 impactOnSkew = amount.divDecimal(skewScale).mulDecimal(self.discountScalar);
+            discount = (
+                MathUtil.max(
+                    MathUtil.min(impactOnSkew, self.lowerLimitDiscount),
+                    self.upperLimitDiscount
+                )
+            );
+        }
+        // first get value of collateral in usd
+        uint128 sellTxnType = 2;
```

Comment:
I believe this is what we discussed the other day, about updating to use the sport price without any pd impact.

Reply:
yeah, we're using the sell feed on the spot market so whatever that oracle returns we use without any skew/util/fixed fees.

Reply:
but what's the hardcoded type 2 there? isn't there a better way to fetch that?

Reply:
Like probably i don't understand this fully, but let's say ETH price is 2000$. So if we specify 1, it returns `1/2000`, if we specify `2` it returns 2000?
And the `1` and `2` are things we made up here to call the right function, in order to point to the buy or sell function. Ok my only concern is that spot price can be pushed around by price impact?  Since I think this should be purely the pyth price, not the price comping from the skew adjusted spot market.